eng/native/configurecompiler.cmake

```diff
@@ -207,12 +215,16 @@ elseif (CLR_CMAKE_HOST_ARCH_ARM)
   add_definitions(-DHOST_ARM)
 elseif (CLR_CMAKE_HOST_ARCH_ARM64)
   set(ARCH_HOST_NAME arm64)
-  add_definitions(-DHOST_ARM64)
-  add_definitions(-DHOST_64BIT)
+  add_definitions(-DHOST_ARM64 -DHOST_64BIT)
 elseif (CLR_CMAKE_HOST_ARCH_S390X)
   set(ARCH_HOST_NAME s390x)
-  add_definitions(-DHOST_S390X)
-  add_definitions(-DHOST_64BIT)
+  add_definitions(-DHOST_S390X -DHOST_64BIT -DBIGENDIAN)
```

Comment:
FYI @uweigand

Reply:
Hmm.  I'm wondering if there is some cross-compilation issue here.  Currently, BIGENDIAN is defined in src/libraries/Native/Unix/CMakeLists.txt based on CLR_CMAKE_TARGET_ARCH_S390X, but in src/coreclr/CMakeLists.txt based on CLR_CMAKE_HOST_ARCH_S390X.  If this is now centralized, do we need to distinguish HOST_BIGENDIAN vs. TARGET_BIGENDIAN like for the other defines?

Reply:
Thinking about this a bit more, it seems this is _currently_ not a problem in practice.   In all current s390x build scenarios, CLR_CMAKE_TARGET_ARCH always equals CLR_CMAKE_HOST_ARCH (even when e.g. cross-building a s390x runtime on an x86 machine).

The only case where CLR_CMAKE_TARGET_ARCH != CLR_CMAKE_HOST_ARCH is when building a cross-compiler component like crossgen2.   None of these are currently supported for s390x.   But if we ever want to enable any of those in the future, endianness differences between target and host could become a point of confusion, and will have to be addressed then.